src/rates/rates.service.ts

```diff
@@ -276,12 +270,23 @@ export class RatesService extends RatesMerger {
       });
     }
 
-    await this.timestampModel.create({
-      date,
-    });
-    await this.tickerModel.create(tickers);
+    try {
+      await this.tickerModel.create(tickers);
+      await this.timestampModel.create({
+        date,
+      });
 
-    this.lastUpdated = date;
+      this.lastUpdated = date;
+
+      this.logger.log(
+        `Rates from ${availableSources}/${this.sourceCount} sources saved successfully`,
+      );
+    } catch (error) {
+      this.fail(
+        `Error: Unable to save new rates in history database: ${error}. See logs for details`,
```

Comment:
Trim possible leading dot from `error`.

`Error: Unable to save new rates in history database: ValidationError: rate: Path rate is required.. See logs for details.` ⟶ `Error: Unable to save new rates in history database: ValidationError: rate: Path rate is required. See logs for details.`